src/IO/S3/ProxyConfigurationResolverAdapter.cpp

```diff
@@ -0,0 +1,58 @@
+#include <IO/S3/ProxyConfigurationResolverAdapter.h>
+
+#if USE_AWS_S3
+
+namespace DB::S3
+{
+
+namespace
+{
+    auto protocolToAWSScheme(DB::ProxyConfiguration::Protocol protocol)
+    {
+        switch (protocol)
+        {
+            case DB::ProxyConfiguration::Protocol::HTTP:
+                return Aws::Http::Scheme::HTTP;
+            case DB::ProxyConfiguration::Protocol::HTTPS:
+                return Aws::Http::Scheme::HTTPS;
+            case DB::ProxyConfiguration::Protocol::ANY:
+                // default to HTTP since there is no ANY in AWS::Scheme and we don't want an exception
+                return Aws::Http::Scheme::HTTP;
+        }
+    }
+
+    auto AWSSchemeToProtocol(Aws::Http::Scheme scheme)
+    {
+        switch (scheme)
+        {
+            case Aws::Http::Scheme::HTTP:
+                return DB::ProxyConfiguration::Protocol::HTTP;
+            case Aws::Http::Scheme::HTTPS:
+                return DB::ProxyConfiguration::Protocol::HTTPS;
+        }
+    }
+}
+
+ClientConfigurationPerRequest ProxyConfigurationResolverAdapter::getConfiguration(const Aws::Http::HttpRequest &)
+{
```

Comment:
The argument is unused and (if I'm not mistaken) the method is not virtual (so we can easily change the signature). Why do we need this argument?

Reply:
Method is virtual, but it is unused indeed. I guess it's there for historical reasons, but we can remove it

Reply:
Sorry, I indeed got confused by `DB::S3::ProxyConfiguration`. It's okay to keep that argument, but we definitely should rename something (`DB::S3::ProxyConfiguration` or `DB::ProxyConfiguration`) to avoid confusion